rust/src/libnm_dbus/connection/bridge.rs

```diff
@@ -16,51 +16,76 @@
 use std::collections::HashMap;
 use std::convert::TryFrom;
 
-use crate::{dbus_value::value_hash_get_bool, error::NmError};
+use crate::{dbus_value::own_value_to_bool, error::NmError};
 
 #[derive(Debug, Clone, PartialEq, Default)]
 pub struct NmSettingBridge {
     pub stp: Option<bool>,
+    _other: HashMap<String, zvariant::OwnedValue>,
 }
 
 impl TryFrom<&HashMap<String, zvariant::OwnedValue>> for NmSettingBridge {
     type Error = NmError;
     fn try_from(
-        value: &HashMap<String, zvariant::OwnedValue>,
+        setting_value: &HashMap<String, zvariant::OwnedValue>,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            stp: value_hash_get_bool(value, "stp")?,
-        })
+        let mut setting = Self::new();
+        for (key, value) in setting_value.iter() {
+            match key.as_str() {
+                "stp" => {
+                    setting.stp = Some(own_value_to_bool(value)?);
+                }
+                _ => {
+                    setting._other.insert(key.to_string(), value.clone());
+                }
+            }
+        }
+        Ok(setting)
     }
 }
 
 impl NmSettingBridge {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
     pub(crate) fn to_value(
         &self,
     ) -> Result<HashMap<&str, zvariant::Value>, NmError> {
         let mut ret = HashMap::new();
         if let Some(v) = self.stp {
             ret.insert("stp", zvariant::Value::new(v));
         }
+        for (key, value) in &self._other {
+            ret.insert(key.as_str(), zvariant::Value::from(value.clone()));
+        }
```

Comment:
Could be:
```
ret.extend(self._other.iter().map(|(key, value)| {
    (key.as_str(), zvariant::Value::from(value.clone()))
}));
```

But I guess here it is more about preference.

Reply:
The code is better. Will change to that.